### ISSUE TITLE
[Enhancement] Allow for multiple `let` `in` statements 

### ISSUE THREAD

**Issue by FireFighter1017** [microsoft/vscode-powerquery#204]
**Is your feature request related to a problem? Please describe.**
When using copy/paste of queries from a Dataflow editor or Power Query editor (select all queries, then ctrl-c, then go in vscode and paste the code) it always report problems for queries referring to other queries in the same clipboard content. 
I use this for making mass changes to queries of a same model/dataflow and/or search through code for a particular token.

**Describe the solution you'd like**
 It would be nice if we could copy/paste queries and that either it formats the code so that VSCode can recognize query names and stop reporting problems in referencing queries .

**Describe alternatives you've considered**
I have tried using the command `powerquery.extractDataflowDocument` for Dataflows, but requires that I export the dataflow into a JSON file first and most of the time I'd like to be able to copy/paste, back and forth, while editing the Dataflow.
And this method does not apply to pbix Datasets.
All that being said, if Power Query editors would let us do find and replace through code, I wouldn't need VS Code at all.

**Additional context**
Here is an example of a script file I copy/pasted from a copy/paste from a dataflow:

```
// WS_DATA_CONVERSION
"some-dataflow-id" meta [IsParameterQuery = true, IsParameterQueryRequired = true, Type = type text]

// DF_PM_1_EXTRACT
"some-dataflow-id" meta [IsParameterQuery = true, IsParameterQueryRequired = true, Type = type text]

// DF_EQUIP
"some-dataflow-id" meta [IsParameterQuery = true, IsParameterQueryRequired = true, Type = type any]

// task-id-generator
let
    Source = PowerBI.Dataflows(),
    Workspace = Source{[workspaceId=WS_DATA_CONVERSION]}[Data],
    Dataflow = Workspace{[dataflowId=DF_PM_1_EXTRACT]}[Data],
    Entity = Dataflow{[entity="task-id-generator"]}[Data]
in
    Entity

// EQ/FLOC/ASSY - Cleaned
let
    Source = PowerBI.Dataflows(),
    Workspace = Source{[workspaceId=WS_DATA_CONVERSION]}[Data],
    Dataflow = Workspace{[dataflowId=DF_EQUIP]}[Data],
    Entity = Dataflow{[entity="EQ/FLOC/ASSY - Cleaned"]}[Data]
in
    Entity
```

The query names are in comments and there is no separation character like ';' between queries which could make it difficult for the language parser to recognize.
Though somehow, when pasting this same clipboard content into a Power Query editor the editor recognizes the query names.  Even folders are recognized (which are not pasted when pasting the clipboard contents into a text editor for some reason...)

**Comment by bgribaudo** [comment]
Would a quick action 'convert query past -> section document' help here? 

(Aside: It would be really neat if Query Editor [desktop] and Power Query Online offered a view/edit section document dialog, where the user could copy out the entire section document and/or paste an edited one back in.)

**Comment by mattmasson** [comment]
> Would a quick action 'convert query past -> section document' help here?

Yes, I can see that being a helpful feature and pretty straightforward for someone to implement.

> All that being said, if Power Query editors would let us do find and replace through code, I wouldn't need VS Code at all.

To me, this stands out as the key point / actual feature people are looking for. I know it's something we've been wanting to do, but I'm not sure where it lands on our overall backlog. 

> offered a view/edit section document dialog

In Desktop, you can go to **Help | About | Copy session diagnostics to clipboard** to get a copy of the section document.

In Power Query Online, I believe exporting the Dataflow.json is currently the only way to see the full section document. However, [the Git integration work for Dataflows Gen2](https://learn.microsoft.com/en-us/fabric/data-factory/dataflow-gen2-cicd-and-git-integration) will also sync out the full section document (in a separate file, so it is not json encoded/escaped).  

In the past we've viewed editing of the section document as high risk, as a corrupted/invalid document can break the PQ UI. I think it's doable with the right checks/validation but would always remain an advanced "use at your own risk" type of feature. Exposing specific UX to support find/replace and other refactoring operations that work across the dataflow might be a safer for most users. Maybe advanced users will be happy with using the git integration to edit their section documents in VS Code.
